test/lib.bash

```diff
@@ -45,23 +45,30 @@ function run_e2e_tests {
     && return 1
 }
 
-function run_knative_serving_tests {
-  (
-  local knative_version=$1
-  # Setup a temporary GOPATH to safely check out the repository without breaking other things.
+# Setup a temporary GOPATH to safely check out the repository without breaking other things.
+# CAUTION: function overrides GOPATH so use it in subshell or restore original value!
+function make_temporary_gopath {
```

Comment:
Why the check if GOPATH exists is important? Okay for putting $GOPATH/bin on PATH

Reply:
Because when it doesn't exist the `cp` command will fail and since we're using `-e` the whole script will fail. We want to re-use the script in downstream (Jenkins) testing where GOPATH might not be available as a variable.

Reply:
That's why I use `$(go env GOPATH)`, instead of `$GOPATH`